web/src/pages/kiosk/[action_id].tsx

```diff
@@ -61,9 +68,17 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
   if (data?.action[0].action === "") {
     return { props: { error_code: "no_sign_in" } };
   }
-
+  const processedAction: ActionKioskType = {
+    ...data.action[0],
+    app: {
+      ...data.action[0].app,
+      app_metadata: data.action[0].app.app_metadata?.[0] || null,
```

Comment:
let's do a cleaner approach here where the backend already handles the logic for what to display, i.e. we only pass a single `app_metadata.name` to the client, a single `app_metadata.is_verified`. if we pass both, the client needs to decide what to display, which will be prone to errors

Reply:
similar to how we have it in precheck